src/uzfs_mgmt.c

```diff
@@ -505,6 +495,66 @@ is_internally_created_clone_volume(const char *ds_name)
 	    REBUILD_SNAPSHOT_CLONENAME));
 }
 
+/*
+ * update_zvol_property updates the zv values
+ * according to given property 'nvprops'
+ */
+int
+update_zvol_property(zvol_state_t *zv, nvlist_t *nvprops)
+{
+	nvpair_t *elem = NULL;
+	nvlist_t *nv = NULL;
+	char *ip = NULL, *replica_id = NULL;
+	char *val = NULL;
+	zpool_prop_t prop;
+	int error = 0;
+
+	while ((elem = nvlist_next_nvpair(nvprops, elem)) != NULL) {
+		prop = zfs_name_to_prop(nvpair_name(elem));
+		val = NULL;
+
+		switch (prop) {
+			case ZFS_PROP_REPLICA_ID:
+			case ZFS_PROP_TARGETIP:
+			case ZFS_PROP_WORKERS:
+				if (nvpair_type(elem) == DATA_TYPE_NVLIST) {
```

Comment:
I guess this will never come, but if that is the case you should should a recursive call because nvlist can also have nvlist.? Lets do assertive coding here, if this is not possible let it fail.

Reply:
if I understood correctly, this gets called from two places - one is in import path and other is in create path.
Import path one seems to be getting props in <nvlist: <nvlist: string>>, where as, create path seems to be getting props in <nvlist: string>.. so, checks for nvlist seems to be required..